### ISSUE TITLE
'ngInject' string literal causes double annotations if file is processed twice

### ISSUE THREAD

**Issue by diwu1989** [olov/ng-annotate#131]
we hit a bug where a file went through ng-annotate twice, and because the 'ngInject' literal doesn't get taken out of the generated file, we ended up with double annotated controllers that don't work

I think after the file goes through annotation, we should strip the 'ngInject' annotations so that there's no double annotation

**Comment by olov** [comment]
Double annotations should not happen - can you please submit a test case?

ngInject annotations are left intact so that they can later be removed and rewritten.

**Comment by diwu1989** [comment]
```js
app.config(["$routeProvider", "localStorageServiceProvider", "hotkeysProvider", function($routeProvider, localStorageServiceProvider, hotkeysProvider) {
  var promise, resolve;
  if (window.KARMA_TEST_RUN) {
    return;
  }
  promise = null;
  return resolve = ["$rootScope", "$q", "Auth", "Notifications", "PageLoader", function($rootScope, $q, Auth, Notifications, PageLoader) {
    'ngInject';
    if (promise) {
      return promise;
    }
    promise = PageLoader.waitFor(Auth).then(function(currentUser) {
      localStorageServiceProvider.setPrefix("bw-" + window.VERSION + "-" + currentUser.id);
      $rootScope.currentUser = currentUser;
      return Notifications.refreshRecentActions();
    });
    return promise;
  }];
}]);
```

when running that through `ng-annotate -a -r annotated.js`
I get 
```js
app.config(["$routeProvider", "localStorageServiceProvider", "hotkeysProvider", function($routeProvider, localStorageServiceProvider, hotkeysProvider) {
  var promise, resolve;
  if (window.KARMA_TEST_RUN) {
    return;
  }
  promise = null;
  return resolve = ["$rootScope", "$q", "Auth", "Notifications", "PageLoader", ["$rootScope", "$q", "Auth", "Notifications", "PageLoader", function($rootScope, $q, Auth, Notifications, PageLoader) {
    'ngInject';
    if (promise) {
      return promise;
    }
    promise = PageLoader.waitFor(Auth).then(function(currentUser) {
      localStorageServiceProvider.setPrefix("bw-" + window.VERSION + "-" + currentUser.id);
      $rootScope.currentUser = currentUser;
      return Notifications.refreshRecentActions();
    });
    return promise;
  }]];
}]);
```

**Comment by olov** [comment]
Thanks. The issue happens for all ngInject forms. Here's the reduced test case:

```js
var x = ["a", ngInject(function(a) { })];
var y = ["a", /*@ngInject*/(function(a) { })];
var z = ["a", function(a) { "ngInject" }];

=>

~/projects/ng-annotate % ./ng-annotate-harmony -a -r tests-issues/issue131.js
var x = ["a", ngInject(["a", function(a) { }])];
var y = ["a", /*@ngInject*/(["a", function(a) { }])];
var z = ["a", ["a", function(a) { "ngInject" }]];
```

Will fix.

**Comment by olov** [comment]
Fixed in 0.15.3. Thanks for reporting!

I also fixed ng-annotate incorrectly adding newlines when rebuilding fn.$inject arrays.

Running `ng-annotate -a` or `ng-annotate -a -r` on a fully annotated file will now produce output identical to the input again. I also added double-add and rebuild tests to guard against similar failures in the future.
